dask/dataframe/core.py

```diff
@@ -5138,17 +5138,20 @@ def cov_corr_chunk(df, corr=False):
         with warnings.catch_warnings(record=True):
             warnings.simplefilter("always")
             mu = (sums / counts).T
-        m = np.zeros(shape)
+        m = np.zeros_like(df.values, shape=shape)
         mask = df.isnull().values
         for idx, x in enumerate(df):
-            # Use .values to get the ndarray for the ufunc.
-            mu_discrepancy = np.subtract.outer(df.iloc[:, idx].values, mu[idx]) ** 2
+            # Avoid using ufunc.outer, since it is not supported by cupy
+            mu_discrepancy = np.zeros_like(df.values, shape=(len(df), len(mu)))
+            for j in range(len(mu)):
+                mu_discrepancy[:, j] = np.subtract(df.iloc[:, idx].values, mu[idx, j])
+            mu_discrepancy = mu_discrepancy ** 2
             mu_discrepancy[mask] = np.nan
             m[idx] = np.nansum(mu_discrepancy, axis=0)
         m = m.T
         dtype.append(("m", m.dtype))
 
-    out = np.empty(counts.shape, dtype=dtype)
+    out = {}
     out["sum"] = sums
     out["count"] = counts
     out["cov"] = cov * (counts - 1)
```

Comment:
Very slight style preference to create the full dictionary in one shot rather than iteratively build it.

```python
return {
    "sum": sums,
    "count": counts,
    "cov": cov * (counts - 1)
}
```

Reply:
This sort of thing happens a few times below.  If it makes sense to you to create single objects like this rather than build them up I would appreciate it a bit.  It's not a big deal though.

My somewhat subjective reason for liking this is that as a future code reviewer I don't have to understand that these lines of code are constructing some output.  When I see a dict literal or a list comprehension my brain is more able to quickly pattern recognize it and move on.

Reply:
Sorry - I totally agree.  I had it this way because I was switching between dict and recarray while debugging, and forgot to clean up